Source: Debian
Upstream: No, upstream is archived

From: Michael Biebl <biebl@debian.org>
Date: Sun, 11 Dec 2022 09:56:59 -0500
Subject: Avoid a circular dependency between -daemon and libcryptui

by statically linking seahorse-daemon to libcryptui.
---
 daemon/Makefile.am     | 2 +-
 libcryptui/Makefile.am | 4 ++++
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/daemon/Makefile.am b/daemon/Makefile.am
index dd8b32e..c1e64e1 100644
--- a/daemon/Makefile.am
+++ b/daemon/Makefile.am
@@ -74,7 +74,7 @@ seahorse_daemon_SOURCES = \
 	$(BUILT_SOURCES)
 
 seahorse_daemon_LDADD = \
-	$(top_builddir)/libcryptui/libcryptui.la \
+	$(top_builddir)/libcryptui/libcryptui-private.la \
 	$(top_builddir)/libegg/libeggdatetime.la \
 	$(top_builddir)/libegg/libeggsmclient.la \
 	$(SEAHORSE_LIBS)
diff --git a/libcryptui/Makefile.am b/libcryptui/Makefile.am
index 76ed5ad..85c8c98 100644
--- a/libcryptui/Makefile.am
+++ b/libcryptui/Makefile.am
@@ -34,6 +34,7 @@ cryptui-marshal.c: cryptui-marshal.list $(GLIB_GENMARSHAL)
 # 
 
 lib_LTLIBRARIES = libcryptui.la
+noinst_LTLIBRARIES = libcryptui-private.la
 
 libcryptui_src = \
     $(srcdir)/cryptui.c \
@@ -55,6 +56,9 @@ libcryptui_la_LDFLAGS = \
     
 libcryptui_la_LIBADD = $(LIBCRYPTUI_LIBS) 
 
+libcryptui_private_la_SOURCES = $(libcryptui_la_SOURCES)
+libcryptui_private_la_LIBADD = $(libcryptui_la_LIBADD)
+
 # ------------------------------------------------------------------
 # PKGCONFIG
 # 
